### PR TITLE
Make Marshal.SizeOf AOT friendly

### PR DESCRIPTION
**Pull request by kant2002**
See https://github.com/dotnet/corert/issues/5374#issuecomment-455706335
Current state in NativeAOT experiment is the same.

### CODE REVIEW

File: Source/Libs/CairoSharp/Context.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ static Context ()
 			// except in the case of Win64 where sizeof(long)
 			// is 32 bits
 			//
-			int ptr_size = Marshal.SizeOf (typeof (IntPtr));
+			int ptr_size = Marshal.SizeOf<IntPtr>();
```
Comment (jkotas) [inline Source/Libs/CairoSharp/Context.cs:68]: Marshal.SizeOf is unnecessary overhead for blittable types. It is better to just use `sizeof(IntPtr)`, sizeof(ulong)`, etc.

**Comment by kant2002** [comment]
Should I directly ask @harry-cpp or somebody else ?

**Comment by knuxbbs** [comment]
> Should I directly ask @harry-cpp or somebody else ?

What does @lytico think? AOT support is very desirable for a library like this.

**Comment by lytico** [comment]
i think we should follow @jkotas advice.
and don't forget code in CapiCodeGen

**Comment by knuxbbs** [comment]
@kant2002 thanks for your involvement with this project. I read about some progress you've made in Windows Forms with NativeAOT.

I hope the maintainers approves this.

**Comment by kant2002** [comment]
@lytico how can I run codegen to look at the output?

@knuxbbs thanks for kind words.

File: Source/Libs/GLibSharp/HookList.cs [hunk 2]
```diff
@@ -12,58 +12,58 @@ namespace GLib {
...
+						, (uint) sizeof (uint) // hook_size
 						, "seq_id"
 						, "is_setup"
 						, 1
 						, 16
 						),
 				new GLib.AbiField("is_setup"
 						, -1
-						, (uint) Marshal.SizeOf(typeof(bool)) // is_setup
+						, (uint) sizeof (bool) // is_setup
```
Comment (jkotas) [inline Source/Libs/GLibSharp/HookList.cs:34]: bool is not blittable type. This should be `sizeof(uint)`, or it should stay as `Marshal.SizeOf`

Comment (lytico) [reply Source/Libs/GLibSharp/HookList.cs]: i would prefer `Marshal.SizeOf<bool>()` who knows if it's always uint on all platforms / cpu architectures ...

Comment (kant2002) [reply Source/Libs/GLibSharp/HookList.cs]: @lytico I'm done that.

**Comment by lytico** [comment]
@kant2002 thanks. i'll test as soon as possible

**Comment by lytico** [comment]
tested. examples give no error.

**Pull request merged**